tests/unit/mesh/test_pmesh.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "mfem.hpp"
+#include "unit_tests.hpp"
+
+namespace mfem
+{
+#ifdef MFEM_USE_MPI
+
+TEST_CASE("ParMeshGlobalIndices",  "[Parallel], [ParMesh]")
+{
+   const int ne = 5;
+
+   for (int dimension = 1; dimension < 4; ++dimension)
+   {
+      Mesh* mesh;
+      if (dimension == 1)
+      {
+         mesh = new Mesh(ne, 1.0);
+      }
+      else if (dimension == 2)
+      {
+         mesh = new Mesh(ne, ne, Element::QUADRILATERAL, 1, 1.0, 1.0);
```

Comment:
The test is hitting a MFEM_VERIFY for amr meshes.

```suggestion
         const char *mesh_file = "../../data/amr-quad.mesh";
         mesh = new Mesh(mesh_file, 1, 1);
```

```
Verification failed: (ldof_ltdof[ldof] >= 0) is false:
 --> ldof 6 not a true DOF.
 ... in function: HYPRE_Int mfem::ParFiniteElementSpace::GetGlobalTDofNumber(int) const
 ... in file: fem/pfespace.cpp:829
```

Reply:
The current implementation assumes a one-to-one correspondence between mesh vertices and DOFs. For AMR meshes, there can be mesh vertices whose DOF is constrained by a neighboring coarse element, meaning it does not have a DOF. In the case of `data/amr-quad.mesh`, there are 8 vertices with no DOF. A more general implementation would be needed for AMR meshes, which is beyond the scope of this PR. This is a good reason to restore the sanity checks that were removed, to catch the AMR case.

Reply:
Could we use `int ParFiniteElementSpace::GetLocalTDofNumber(int ldof) const` ?. This returns `-1` if the `ldof` is not owen or it's a slave.

Reply:
@psocratis That was the first thing I tried, but this fails even with one MPI task. It is not a matter of the DOF being owned by another process; there is not even a DOF for the vertex at the midpoint of a neighboring coarse edge. The element has a DOF, but the `ParFiniteElementSpace` does not have a DOF (it is constrained and eliminated, as I understand it). For `data/amr-quad.mesh`, there are 41 vertices but only 33 DOFs (so my original sanity check would fail).

Reply:
I see, some of those checks were not so excessive. Isn't the below one sufficient?
```
   MFEM_VERIFY(fespace.GetVSize() == GetNV(), "There should be a bijection between DOFs and vertices");
```

The doxygen should state which of the functions work with AMR meshes and which don't.

Reply:
My comment above was slightly wrong. There are 41 vertices and DOFs, with one DOF per vertex, but only 33 true DOFs, with one MPI process. So the original sanity checks pass, and the failure is on a verify in pfespace.cpp. This implementation will not work for AMR vertices, edges, and faces, because they have one DOF which is not a true DOF on any process, and there is no simple way to get a global index from the `ParFiniteElementSpace`. A different approach will be needed in the future, to support AMR. Getting global indices for elements does work. I documented whether each function supports AMR, and I removed the sanity checks (since they don't actually catch the AMR case).